pkg/config/options.go

```diff
@@ -34,11 +42,51 @@ func (o *Options) Validate() error {
 	return nil
 }
 
+func (o *Options) Complete() error {
+	if o.onlyProcessChanges {
+		o.shouldProcessChanges = true
+		mainBranch := getUpstreamBranch()
+
+		cmd := exec.Command("git", "diff", "--name-status", mainBranch)
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("couldn't get the changed files comparing with %s branch: %w", mainBranch, err)
+		}
+
+		trimmedLastLine := strings.TrimSuffix(string(out), "\n")
+		changedFilesWithStatus := strings.Split(trimmedLastLine, "\n")
+
+		// Fill the modified files set based on the git output. If we detect anything else than modified files,
+		// make shouldProcessChanges = false, in order to process all the files instead of a subset. For now we
+		// care of processing on modified files when there are ONLY modified files.
+		// Example output of `git diff --name-status master`
+		// M       path/of/modified/file.yaml
+		// D       path/of/deleted/file.yaml
+		// R100    path/of/renamed/old-name.yaml   path/of/renamed/new-name.yaml
+		o.modifiedFiles = sets.NewString()
+		for _, f := range changedFilesWithStatus {
+			statusAndFile := strings.Fields(f)
+			if statusAndFile[0] == "M" {
+				o.modifiedFiles.Insert(statusAndFile[1])
+				continue
+			}
+			o.shouldProcessChanges = false
```

Comment:
Why do we care to track two booleans? In what case do we want to know that we were asked to only process changes but will be doing everything?

Reply:
well, the user can ask to process only the changes, but if we find that there are non `M` changes we should skip it.

Reply:
I understand that, but why do we need anything more than a single boolean to track whether we will or will not process everything? Set it to true when the user asks for it, set it to false when we detect non-`M` changes.

Reply:
addressed